File: javascript/rna-transcription/rna-transcription.js
//
// This is only a SKELETON file for the 'RNA Transcription' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

//input string
//export string
export const toRna = (str = "") => {
  
  const pairs = {
    "A": "U",
    "T": "A",
    "C": "G",
    "G": "C"
  }

  return str.split('').map( dna => pairs[dna]).join('');

};

File: ruby/rna-transcription/rna_transcription.rb
class RnaTranscription

  def self.of_dna(dna)
    hash = {'G'=>'C', 'C'=>'G', 'T'=>'A', 'A'=>'U'}
    arr = dna.split(//)

   arr.map do |dna_code|
    hash[dna_code]
   end.join
  end
end